CRM/Event/Form/SelfSvcUpdate.php

```diff
@@ -123,8 +115,7 @@ public function preProcess() {
     $this->_event_id = $this->_part_values['event_id'];
     $url = CRM_Utils_System::url('civicrm/event/info', "reset=1&id={$this->_event_id}");
     $this->_contact_id = $this->_part_values['participant_contact_id'];
-    $validUser = CRM_Contact_BAO_Contact_Utils::validChecksum($this->_contact_id, $this->_userChecksum);
-    if (!$validUser && !CRM_Core_Permission::check('edit all events')) {
+    if (!$this->getAuthenticatedCheckSumContactID() && !CRM_Core_Permission::check('edit all events')) {
       CRM_Core_Error::statusBounce(ts('You do not have sufficient permission to transfer/cancel this participant.'), $url);
```

Comment:
I'm getting this bounce error and it works before the patch, so something is off with this refactor.

Reply:
getRequestedContactID() is returning NULL

Reply:
This PR has conflicts now too, so will close.